src/server.c

```diff
@@ -2441,6 +2441,12 @@ void initServer(void) {
     server.reply_buffer_resizing_enabled = 1;
     resetReplicationBuffer();
 
+    /* Make sure the locale is set on startup based on the config file. */
+    if (setlocale(LC_COLLATE,server.locale_collate) == NULL) {
+        serverLog(LL_WARNING, "Failed to configure LOCALE. %s", strerror(errno));
```

Comment:
```suggestion
        serverLog(LL_WARNING, "Failed to configure LOCALE: %s", strerror(errno));
```

Reply:
In fact I found that the message `errno` carries was not so proper to show here. Maybe we could replace it.
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/56780191/185368743-bab1e1e2-4f84-4b90-a071-04142a92dc55.png">

Reply:
usually, on system errors, i'll argue that it's ok.. i.e. the people looking at this will be quick to realize that's ENOENT, and that they should look setlocale man page (it's far better to have that info, than not have it).
btw, i don't see any reference to errno in the man page at all!?

however, in this case if it's a plain user typo, or attempting to use a locale that's not supported, maybe we should print something more friendly.
i suppose it should be the same message we return in CONFIG SET command for an unsupported input (in which we didn't print the errno).

so maybe we should handle ENOENT separately from other errors?
can you check if this API is at all using errno?

Reply:
I've browsed the source code of `setlocale.c`. There's indeed usage of `errno` but it seems that only EINVAL is used in this and other related APIs ( e.g. ` _nl_find_locale()`, not quite sure about some implicit references, see https://code.woboq.org/userspace/glibc/locale/setlocale.c.html#setlocale), and I'm uncertain when the value of `errno` was assigned to ENOENT. I agree to print something more intuitive, maybe like `serverLog(LL_WARNING, "Failed to configure LOCALE for invalid locale name.");`

Reply:
i see it was discussed here: https://stackoverflow.com/questions/33668695/setlocale-raising-errno-while-a-non-null-value-is-returned and the conclusion is that setlocale doesn't (or shouldn't) mess with errno.
let's drop using it and replace it with the string you suggested.

Reply:
Ok, I'll replace then.